### PR TITLE
Task sf/j368 validate query for user profile

### PR DESCRIPTION
**Pull request by Sayonarko**

### CODE REVIEW

File: backend/src/middlewares/check-params-is-valid/check-params-is-valid.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { HttpError } from '~/exceptions/exceptions';
+import { ErrorMessage, HttpCode } from '~/common/enums/enums';
+
+const checkParamsIsValid = (): RequestHandler => {
+  const handler: RequestHandler = (req, _res, next) => {
+
+    const params = Object.values(req.params);
+
+    params.forEach((param) => {
+      if (isNaN(Number(param))) {
```
Comment (artem-totality) [inline backend/src/middlewares/check-params-is-valid/check-params-is-valid.middleware.ts]: What about id = 2147483648 and greater? 😉

**Review by artem-totality — DISMISSED** [review]

File: backend/src/middlewares/check-params-is-valid/check-params-is-valid.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { RequestHandler } from 'express';
+import { HttpError } from '~/exceptions/exceptions';
+import { ErrorMessage, HttpCode } from '~/common/enums/enums';
+import { MAX_POSSIBLE_ID } from '~/common/constants/constants';
+
+const checkParamsIsValid = (): RequestHandler => {
```
Comment (what1s1ove) [inline backend/src/middlewares/check-params-is-valid/check-params-is-valid.middleware.ts]: but we can have not only id in parameters
we should create something more specific for it

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/middlewares/check-params-is-valid/check-params-is-valid.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { RequestHandler } from 'express';
+import { HttpError } from '~/exceptions/exceptions';
+import { ErrorMessage, HttpCode, RouterParamsId } from '~/common/enums/enums';
+import { MAX_POSSIBLE_ID } from '~/common/constants/constants';
+
+const checkParamsIsValid = (): RequestHandler => {
+  const handler: RequestHandler = (req, _res, next) => {
+    const params = Object.values(RouterParamsId);
```
Comment (what1s1ove) [inline backend/src/middlewares/check-params-is-valid/check-params-is-valid.middleware.ts]: do it as in the `checkAuthMiddleware` (pass this params through function params)

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/common/enums/express/router-params.enum.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+enum RouterParams {
```
Comment (what1s1ove) [inline backend/src/common/enums/express/router-params.enum.ts]:
```suggestion
enum RouterParam {
```
enum should be always singular

File: backend/src/middlewares/check-params-is-valid/check-params-is-valid.middleware.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import { MAX_POSSIBLE_ID } from '~/common/constants/constants';
+
+const checkParamsIsValid = (...params: RouterParams[]): RequestHandler => {
+  const handler: RequestHandler = (req, _res, next) => {
+
+    params.forEach((param) => {
+      const checkParam = req.params[param];
+
+      if (checkParam) {
+        if (isNaN(Number(checkParam)) || Number(checkParam) > MAX_POSSIBLE_ID) {
```
Comment (what1s1ove) [inline backend/src/middlewares/check-params-is-valid/check-params-is-valid.middleware.ts]:
```suggestion
      if (checkParam && isNaN(Number(checkParam)) || Number(checkParam) > MAX_POSSIBLE_ID) {
```
and add variable/variables for it

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**